src/lib/storage/chunk.cpp

```diff
@@ -1,31 +1,51 @@
 #include "chunk.hpp"
 
-#include "abstract_segment.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  _columns.push_back(segment);
+}
+
+template <typename T>
+bool append_with_type(std::vector<std::shared_ptr<AbstractSegment>>::iterator column_it,
+                      std::vector<AllTypeVariant>::const_iterator value_it) {
+  auto segment = std::dynamic_pointer_cast<ValueSegment<T>>(*column_it);
+  if (segment) {
+    segment->append(*value_it);
+    return true;
+  }
+  return false;
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == column_count(), "Cannot insert a tuple with less values than columns.");
+
+  auto column_it = _columns.begin();
+  auto value_it = values.begin();
+
+  for (; column_it < _columns.end(); ++column_it) {
+    (append_with_type<int32_t>(column_it, value_it) || append_with_type<int64_t>(column_it, value_it) ||
+     append_with_type<float>(column_it, value_it) || append_with_type<double>(column_it, value_it) ||
+     append_with_type<std::string>(column_it, value_it));
+
+    ++value_it;
+  }
 }
 
 std::shared_ptr<AbstractSegment> Chunk::get_segment(const ColumnID column_id) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _columns[column_id];
 }
 
 ColumnCount Chunk::column_count() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return static_cast<ColumnCount>(_columns.size());
```

Comment:
Why is the `static_cast` needed here?

Reply:
_columns.size() is of type `long unsigned int`, whereas ColumnCount is a `uint16_t`. Our compiler complains if we try to build `return _columns.size()`.

Reply:
Ah interesting, I tried it without the `static_cast` and it was no problem. But then I guess it is better to have it